api/src/pcapi/celery_tasks/tasks.py

```diff
@@ -1,20 +1,54 @@
+from contextlib import contextmanager
+import datetime
 from functools import wraps
 import json
 import logging
 import typing
 
+from celery import Task
 from celery import shared_task
+from flask import current_app as app
 import pydantic.v1 as pydantic_v1
 
 
 logger = logging.getLogger(__name__)
 
 
+class RateLimitedError(Exception):
+    def __init__(self, period: int, max_per_period: int, current: int):
+        self.period = period
+        self.max_per_period = max_per_period
+        self.current = current
+
+
+def get_key(name: str, period: int) -> str:
+    # The period_id is the number of period since UNIX epoch
+    # If period is 60 (a minute) and period id is 29061383. It means we are
+    # at the 29061383th minute since epoch.
+    period_id = int(datetime.datetime.utcnow().timestamp()) // period
```

Comment:
Pourquoi ne pas utiliser directement `time.time()` plutôt que créer des objets `datetime` ?
```suggestion
    period_id = int(time..time()) // period
```

Reply:
C'est corrigé